src/Symfony/Component/Routing/Requirement/Requirement.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Requirement;
+
+/**
+ * A collection of universal regular-expression constants to use as route parameter requirements.
+ */
+enum Requirement
+{
+    public const DATE = '[0-9]{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])';
```

Comment:
What about being more explicit in the constant name about the expected format?

```php
public const DATE_YMD = '[0-9]{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])';
```

Reply:
Maybe a comment to show that year must be 4 digits, month/day 2 and separated by hyphens?

```php
public const DATE_YMD = '[0-9]{4}-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])'; // YYYY-MM-DD
```